### PR TITLE
THRIFT-2994: Fix TJsonProtocol for nodejs

### PR DESCRIPTION
**Pull request by timse**
this fixes: https://issues.apache.org/jira/browse/THRIFT-2994 (not sure if that is the correct way to link a jira ticket :smile: )

It initializes the arrays used to build up the json on construct and writes to transport whenever a `writeEnd` method is encountered and the stack length is 1, assuming its finished now.

### CODE REVIEW

**Comment by timse** [comment]
any news on that?

**Comment by timse** [comment]
i did changes, i don't really know what to do with the failing, tests. Are they related to my changes?

**Comment by timse** [comment]
any news on this @RandyAbernethy

File: lib/nodejs/lib/thrift/json_protocol.js [hunk 1]
```diff
@@ -89,9 +89,16 @@ TJSONProtocol.RType.set = Type.SET;
 TJSONProtocol.Version = 1;
 
 TJSONProtocol.prototype.flush = function() {
+  this.writeToTransportIfStackIfFlushable();
```
Comment (nsuke) [inline lib/nodejs/lib/thrift/json_protocol.js]: Do you mean IfStack**Is**Flushable ?

Comment (timse) [inline lib/nodejs/lib/thrift/json_protocol.js]: oups! nice catch, sure. Will change. Thanks :)!

Comment (timse) [inline lib/nodejs/lib/thrift/json_protocol.js]: done :)

**Comment by timse** [comment]
anything else needed for this to proceed @nsuke ?
not sure what to do with the one failing test/if it has anything to do with these changes

**Comment by nsuke** [comment]
For that matter, even a single minimalistic test would be very useful to verify this and also to prevent future regression.
Either way, I think I can look into it soon.

**Pull request closed without merging**